output/output.hpp

```diff
@@ -31,7 +31,9 @@ class Output
 		FLAG_RESTART = 2
 	};
 	virtual void outputBuffer(void *mem, size_t size, int64_t timestamp_us, uint32_t flags);
+	virtual void timestampReady();
 	VideoOptions const *options_;
+	FILE *fp_timestamps_;
 
```

Comment:
One thing I might change here is to make timestampReady() and fp_timestamps_ protected members rather than public.  That way, the derived classes can still access them, but the application cannot.